pkg/autoscaler/collector.go

```diff
@@ -165,6 +172,28 @@ func (c *MetricCollector) Update(ctx context.Context, metric *av1alpha1.Metric)
 	return nil, k8serrors.NewNotFound(av1alpha1.Resource("Metrics"), key.String())
 }
 
+// CreateOrUpdate either creates a collection for the given metric or update it, should
+// it already exist.
```

Comment:
Do we care to do the same two-step locking as we did for breakers?

Reply:
Hmm, maybe? This would be called ~every second for each revision that‘s scraped so by far not as heavily trafficked.

Reply:
But you know -- "ugly but uniform" :-)